### PR TITLE
refactor(typescript): allow to write typescript code

### PR DESCRIPTION
**Pull request by ghusse**
Linked to CU-2wuhpcj

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,10 +93,11 @@
...
     "semantic-release-slack-bot": "3.5.2",
     "sinon": "9.0.2",
-    "supertest": "4.0.2"
+    "supertest": "4.0.2",
+    "typescript": "4.8.4"
   },
   "scripts": {
-    "build": "babel src -d dist",
+    "build": "tsc && babel src -d dist --extensions '.ts,.js'",
     "build:watch": "babel --source-maps inline --watch --out-dir dist src",
```
Comment (Thenkei) [inline package.json:101]: Is this still working?

Comment (Thenkei) [reply package.json:101]: We are missing type checking as it is?

Comment (Thenkei) [inline package.json:101]:
```suggestion
    "build:watch": "babel --source-maps inline --watch --extensions '.ts,.js' --out-dir dist src",
```

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
+{
+  "compilerOptions": {
```
Comment (Thenkei) [inline tsconfig.json:2]: Looking at this resource I'm a little bit lost on the current proposed settings.
https://www.typescriptlang.org/docs/handbook/babel-with-typescript.html#type-checking-and-dts-file-generation

**Review by Thenkei — DISMISSED** [review]
- ✅  Code review - 👌 
- ✅  Automatic tests
  - ✅  Unit tests
- ✅  Manual tests - It works well. I was just preoccupied by the `yarn build:watch` command but it's seems ok.
- ✅  PR title
- ❔  Destination branch - Shouldn't it be the starting point for the next major release?

File: package.json (hunk 1 above)
Comment (ghusse) [reply package.json]: Yes, thanks for the suggestion.

**Review by Thenkei — APPROVED** [review]

**Pull request merged**